bhyve.pyx

```diff
@@ -1,4 +1,107 @@
 cimport bhyve
+cimport types
+cimport vmm
 
+import enum
 
+from libc.errno cimport errno
+from libc.stdlib cimport free
 
+
+class Error(enum.IntEnum):
+    OPEN_FAILED = 1
+    STATS_FETCH_FAILED = 2
+
+
+class BhyveException(RuntimeError):
+    def __init__(self, code, message):
+        super(BhyveException, self).__init__(f'[Errorno {code}] {message}')
+        self.code = code
+
+cdef class VM:
+
+    cdef str vm_name
+    cdef bhyve.vmctx* vm
+
+    def __cinit__(self, str vm_name_p, bint create_vm=0):
+        self.vm_name = vm_name_p
+        vm_name_bytes = vm_name_p.encode()
+        cdef char* vm_name = vm_name_bytes
+
+        if create_vm:
+            with nogil:
+                code = bhyve.vm_create(vm_name)
+
+            if code != 0:
+                raise BhyveException(
+                    errno, f'Failed to create VM {self.vm_name}'
+                )
+
+        with nogil:
+            self.vm = bhyve.vm_open(vm_name)
+
+        if self.vm == NULL:
+            raise BhyveException(
+                Error.OPEN_FAILED, f'Could not open VM {self.vm_name}'
+            )
+
+    def force_reset(self):
+        with nogil:
+            error = bhyve.vm_suspend(self.vm, vmm.VM_SUSPEND_RESET)
+
+        if error:
```

Comment:
It seems libvmmapi simply returns errno from vmm ioctl, which usually are errno.h codes?
Perhaps we could also return the error string from errno?

Reply:
That is a good idea and would increase the understanding of the underlying issue without having to google the description. I couldn't find relevant error strings in the header file, I think I'll have to make a dictionary or something like that for them. Do you think that would make sense ?

Reply:
I imagine you can use strerror()